src/browser/starter.js

```diff
@@ -698,16 +698,31 @@ V86Starter.prototype.run = async function()
  */
 V86Starter.prototype.stop = async function()
 {
-    this.bus.send("cpu-stop");
+    // Emulation is not guaranteed to stop immediately in the same event loop
+    // cycle.
+    // If the emulator is running, this promise will resolve when the emulator
+    // has finished stopping.
+    // Else, this promise will resolve immediately.
+    const promise = new Promise(resolve => {
+        if (!this.cpu_is_running) return resolve();
+        const listener = () => {
+            this.remove_listener("emulator-stopped", listener);
+            resolve();
+        };
+        this.add_listener("emulator-stopped", listener);
+        this.bus.send("cpu-stop");
+    })
+
+    await promise;
```

Comment:
```suggestion
    if(!this.cpu_is_running)
    {
        return;
    }

    const stopped = new Promise(resolve => {
        const listener = () => {
            this.remove_listener("emulator-stopped", listener);
            resolve();
        };
        this.add_listener("emulator-stopped", listener);
    });

    this.bus.send("cpu-stop");

    await stopped;
```

(I suggest removing the comment, since it expresses the same idea as the code itself. For reference as to why the code is written this way, one can always refer to `git blame`).

Reply:
Ah, right, that is much nicer :)

Would you mind if I move `this.bus.send("cpu-stop");` to line 712, so it's clearly guaranteed to be called immediately after `this.add_listener("emulator-stopped", listener);` within the same anonymous function?

In other words, if I'm debugging some complex race condition, there would be no reason to question whether side effects triggered by `"cpu-stop"` have taken effect before the anonymous function passed into `new Promise()` has been called.

(There's also a good chance that I'm overthinking this, as I have been experimenting with alternative JS runtimes lately 😅)

Reply:
To save time, I went ahead and pushed a commit that applies your suggested changes, and a separate commit that applies my aforementioned modification (moving `this.bus.send("cpu-stop");`). The last one can be reverted if you prefer to stick with the suggested changes :)

Reply:
It's probably easier to reason about, indeed.